portfolio/src/main/webapp/scripts/authentication.js

```diff
@@ -0,0 +1,53 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const COMMENTS_NO_ACCESS = 'comment-no-access';
+const FORM_INPUTS = "form-inputs";
+const COMMENT_SUBMIT_BTN = "comment-submit";
+const LOGIN_BTN = "login-status-btn";
+const LOGIN_LINK = "login-link";
+const DISPLAY_SHOW = "inline";
+const DISPLAY_HIDE = "none";
+
+window.addEventListener('load', function() {
+  console.log("authentication.js");
+  fetchLogInStatus();
+});
+
+function fetchLogInStatus() {
+  fetch("/auth-check").then(response => response.json()).then((loginStatus) => {
```

Comment:
Here you would need to use the JSON.parse() function to get a javascript object which will have fields with names like in the Java object. You won't need to predefine that object. So I expect it to be something like
```
fetch("/auth-check").then(response => JSON.parse(response.json())).then((loginStatus) => {
    ...
    if (loginStatus.loggedIn) {
     ...
    }
...
}
```

https://www.w3schools.com/js/js_json_parse.asp

Reply:
I believe that the original statement actually parsed the json already, since the function worked without me needing to use JSON.parse()

Reply:
Well, as long as it works. :)

I didn't know if you needed to handle the object create different from array creation.